rad/src/lib.rs

```diff
@@ -201,11 +208,81 @@ pub async fn run_retrieval_report(
 /// Run retrieval stage of a data request, return `RadonTypes`.
 pub async fn run_retrieval(retrieve: &RADRetrieve) -> Result<RadonTypes> {
     // Disable all execution tracing features, as this is the best-effort version of this method
-    run_retrieval_report(retrieve, RadonScriptExecutionSettings::disable_all())
+    run_retrieval_report(retrieve, RadonScriptExecutionSettings::disable_all(), None)
         .await
         .map(RadonReport::into_inner)
 }
 
+/// Run retrieval using multiple transports, and only produce a positive result if the retrieved
+/// values pass the filter function from the tally stage.
+///
```

Comment:
Is there a reason to use the filter with the tally script instead of the aggregation script? I am assuming the first will be more restrictive and that is the idea, but it also sounds counter-intuitive to have a filter for different sources and apply a different filter to the same source.

Reply:
The tally filter is the filter that will slash the nodes, so one of the risky cases would be a data request with a very strict filter in tally. So it has to be tested to declare the request as "safu"